src/TiledArray/math/linalg/non-distributed/cholesky.h

```diff
@@ -21,38 +21,34 @@
  *  Created:    16 October, 2020
  *
  */
-#ifndef TILEDARRAY_ALGEBRA_LAPACK_CHOL_H__INCLUDED
-#define TILEDARRAY_ALGEBRA_LAPACK_CHOL_H__INCLUDED
+#ifndef TILEDARRAY_MATH_LINALG_NON_DISTRIBUTED_CHOL_H__INCLUDED
+#define TILEDARRAY_MATH_LINALG_NON_DISTRIBUTED_CHOL_H__INCLUDED
 
-#include <TiledArray/algebra/lapack/lapack.h>
-#include <TiledArray/algebra/lapack/util.h>
 #include <TiledArray/config.h>
-#include <TiledArray/conversions/eigen.h>
 
-namespace TiledArray {
-namespace lapack {
+#include <TiledArray/math/linalg/util.h>
+#include <TiledArray/conversions/eigen.h>
+#include <TiledArray/math/linalg/rank-local.h>
 
-namespace detail {
+namespace TiledArray::math::linalg::non_distributed {
 
 template <typename Tile, typename Policy>
-auto make_L_eig(const DistArray<Tile, Policy>& A) {
+auto rank_local_cholesky(const DistArray<Tile, Policy>& A) {
   using Array = DistArray<Tile, Policy>;
   using numeric_type = typename Array::numeric_type;
   static_assert(std::is_same_v<numeric_type, typename Array::element_type>,
                 "TA::lapack::{cholesky*} are only usable with a DistArray of "
                 "scalar types");
 
   World& world = A.world();
-  auto A_eig = detail::to_eigen(A);
+  auto A_eig = detail::make_matrix(A);
   if (world.rank() == 0) {
-    lapack::cholesky(A_eig);
+    linalg::rank_local::cholesky(A_eig);
   }
   world.gop.broadcast_serializable(A_eig, 0);
```

Comment:
Cholesky can be safely replicated with identical inputs (not like XSYEV which has a phase problem), and since `make_matrix` already seems to be replicating the data, that would save you some communication / synchronization